### PR TITLE
Now Adding 404 Page!

### LINKED ISSUE #346: Feature Request — 404 page for non matching routes
**Job**
When user tries to go to a route that does not exist, we should show a 404 Page Not Found error page

**Describe a possible solution**
Have a default route that shows 404 page

**Context**

### PR DESCRIPTION
**Pull request by RichtXO**
**Issue**

Usually, when a user goes into the wilderness of the website, there was no page to alert them that they were lost, alone, and stuck forever. Now they can recover, come back to the main homepage and be free!

> closes #346 

**Test Procedure**

> 1. Go to website
> 2. Add `/asdfasdf` at the end of the URL (Or whatever you want to add at the end, shouldn't matter)
> 3. The 404-page should be there!

**Photos**
In light mode:
image

In dark mode:
image

**Additional Info**
Would like any help to make it look more appealing! Thanks!

### CODE REVIEW

**Comment by marchdan** [comment]
@RichtXO No SIS man? https://github.com/YACS-RCOS/yacs/blob/master/web/src/assets/images/sismanhigh.svg

**Comment by RichtXO** [comment]
> @RichtXO No SIS man? https://github.com/YACS-RCOS/yacs/blob/master/web/src/assets/images/sismanhigh.svg

Crap! I forgot!!!

**Comment by marchdan** [comment]
> > @RichtXO No SIS man? https://github.com/YACS-RCOS/yacs/blob/master/web/src/assets/images/sismanhigh.svg
> 
> Crap! I forgot!!!

Here is the old PR as well if you want to reference it: https://github.com/YACS-RCOS/yacs/pull/490

File: src/web/src/components/CourseList.vue [hunk 1]
```diff
@@ -108,20 +108,16 @@ export default {
...
     return {
       faInfoCircle,
       DAY_SHORTNAMES,
       textSearch: "",
       selectedSubsemester: null,
       selectedDepartment: null,
-      departmentOptions: [{ text: "All", value: null }],
-      courseList: this.courses,
+      // departmentOptions: [{ text: "All", value: null }],
+      // courseList: this.courses,
```
Comment (jshom) [inline src/web/src/components/CourseList.vue:119]: Lets remove the commented lines, its in version control so ez revertible

**Review by jshom — APPROVED** [review]

File: src/web/src/pages/NotFound.vue [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  margin: 0px auto 20px;
+  z-index: -1;
+}
+
+.notfound .notfound-404 h1 {
+  font-family: 'Montserrat', sans-serif;
+  font-size: 236px;
+  font-weight: 200;
+  margin: 0px;
+  color: #211b19;
```
Comment (Viyerelu23333) [inline src/web/src/pages/NotFound.vue:67]: The *Oops* text has a relatively low contrast in dark mode.  
I think removing this `color` to follow text color definitions in `dark.scss` will resolve this problem.

Also, not sure about if `Montserrat` is a safe font face in systems.  It seemed that my device fell back to default sans-serif font...

**Comment by RichtXO** [comment]
@marchdan :)

**Comment by marchdan** [comment]
> @marchdan :)

He's not showing on mobile? 😢

File: src/web/src/pages/NotFound.vue [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  opacity: 0;
+  width: 750px;
+  height: 600px;
+  top: -500px;
+  left: -100px;
+  transition: opacity .5s;
+}
+
+h2:hover img {
+  opacity: 0.1;
```
Comment (marchdan) [inline src/web/src/pages/NotFound.vue:48]: I wonder if he should be ever so slightly darker? Seems a little light even for an easter egg

**Comment by marchdan** [comment]
I love how it's a hover on PC, but on mobile you have to click for it, makes it even more hidden 😏

**Comment by RichtXO** [comment]
> I love how it's a hover on PC, but on mobile you have to click for it, makes it even more hidden 😏

Definitely intended! I thought of that throughout this process, 100%. I did it on purpose :)))

**Review by Viyerelu23333 — APPROVED** [review]

**Pull request merged**